root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs

```diff
@@ -55,14 +63,16 @@
 
 namespace Touryo.Infrastructure.Framework.AsyncProcessingService
 {
-
     /// <summary>
     /// Asynchronous Processing Service class for windows service
     /// </summary>
     public class AsyncProcessingService : System.ServiceProcess.ServiceBase
     {
+        private Thread _mainthread = null;
+        private Thread _workerThread = null;
```

Comment:
Why there is a need two threads of _mainthread  and _workerThread?
- The following thread required for Asynchronous processing service.
  - One of the main thread
  - Multiple worker threads in the thread pool
- This program is redundant because it uses the following threads.
  - One of the main thread
  - One of the worker thread
  - Multiple worker threads in the thread pool